### PR TITLE
feat: Removing missing keys from files

### PR DESCRIPTION
**Pull request by last-partizan**
Here is my first attempt at removing missing keys.

I had to refactor `writeMissingToLanguage` into `LanguageFileUpdater` (is is needed to avoid duplicated reading/writing for adding and deletion of keys), and created separate `FileAccessLayer` to move file-writing into separate entity (probably reading should also be there, but that's for another time).

Refs #113

### CODE REVIEW

**Comment by Spittal** [comment]
Hey thanks! I will review this shortly and hopefully get it added soon.

**Comment by last-partizan** [comment]
Also i removed JSON.stringify from `readLangFiles`, and `JSON.parse` from other places in that file. I hope it wasn't needed for some black magick? :)

**Comment by last-partizan** [comment]
@Spittal i worry i had refactored it too much :) any suggestions is welcomed

**Comment by Spittal** [comment]
It's all good! I have a plan to give this project some much needed updates this week and this code is a big help!

File: tests/unit/report-command/language-files.spec.ts [hunk 1]
```diff
@@ -33,8 +33,14 @@ describe('file: report-command/language-files', () => {
 
     it('Create fails if language files are not valid JSON', () => {
```
Comment (last-partizan) [inline tests/unit/report-command/language-files.spec.ts:34]: btw, it doesn't looks like this is fails.

This is incorrect description or am i missing something?

Comment (Spittal) [reply tests/unit/report-command/language-files.spec.ts:34]: Yeah, that's going to my first bit of business— is getting all the tests written for the new use cases people have brought up for the tool and making sure that the existing tests still all work and test the right conditions.

**Comment by pixari** [comment]
Hi @Spittal! Should we merge this PR or wait for your big update?

**Comment by Spittal** [comment]
Hey! Lets merge this in now, and I'll mix it in my big PR. Thanks @pixari I'll leave the Merge to you 🎉

**Comment by pixari** [comment]
Great, @Spittal!

Thank you for your effort @last-partizan!

**Pull request merged**